src/core/server/metrics/metrics_service.test.ts

```diff
@@ -14,16 +14,28 @@ import { MetricsService } from './metrics_service';
 import { mockCoreContext } from '../core_context.mock';
 import { httpServiceMock } from '../http/http_service.mock';
 import { loggingSystemMock } from '../logging/logging_system.mock';
-import { take } from 'rxjs/operators';
+import { analyticsServiceMock } from '../mocks';
+import { firstValueFrom, Observable } from 'rxjs';
+import { OpsMetrics } from './types';
 
 const testInterval = 100;
 
 const dummyMetrics = { metricA: 'value', metricB: 'otherValue' };
 
 const logger = loggingSystemMock.create();
 
+const createNextEmission = async (getOpsMetrics$: () => Observable<OpsMetrics>) => {
+  jest.advanceTimersByTime(testInterval);
```

Comment:
To make `rxjs` timers work in `jest`, you might need to use `fakeScheduler` from `rxjs-marbles/jest`. Refer to https://github.com/elastic/kibana/blob/722698275900fca36a7fd0d8ec253028e434af8d/packages/analytics/client/src/analytics_client/analytics_client.test.ts#L350-L360 for an example of how to use them